src/Form/EventListener/AnyNoneFilterListener.php

```diff
@@ -45,26 +48,44 @@ class AnyNoneFilterListener
      */
     protected $columnsExtractor;
 
+    /**
+     * Elements (listing entities) containing AnyNoneSelection
+     *
+     * @var string[]
+     */
+    private $entityElementsWithAnyNone = array();
+
     public function __construct(ColumnsExtractor $columnsExtractor)
     {
         $this->columnsExtractor = $columnsExtractor;
     }
 
+    /**
+     * Adds data for AnyNoneFilter to the form (events, ...).
+     *
+     * @param \CubeTools\CubeCommonBundle\Form\EventListener\FormBuilderInterface $builder
+     */
+    public function addToBuilder(FormBuilderInterface $builder)
+    {
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, array($this, 'addAnyNoneColumns'));
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, array($this, 'processAnyNoneColumns'));
```

Comment:
The idea of having it directly in form was to have clear view what event listeners are used for this form (before other event listeners were in form).

Reply:
But when we want to use different events, we have to update all places where the listener is used. Form me the documentation of this function looks like the better place to show this.

Reply:
https://github.com/EmchBerger/cube-common-bundle/pull/115#issuecomment-445212669 - maybe after changes EventSubscriber would be more appropriate solution